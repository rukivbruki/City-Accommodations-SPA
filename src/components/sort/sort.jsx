import React from "react";
import {connect} from "react-redux";
import Option from "../option/option";
import {SortOptions} from "../../constants";
import {ActionCreator} from "../../reducer/data/data";
import PropTypes from "prop-types";

class Sort extends React.PureComponent {
  render() {
    const {
      currentSortType,
      isOpen,
      labelName,
      onOpenSorting,
      onSelectOption,
      onSetTypeSort,
    } = this.props;

    return (
      <>
        <span
          onClick={onOpenSorting}
          className="places__sorting-type"
          tabIndex="0"
        >
          {labelName}
          <svg className="places__sorting-arrow" width="7" height="4">
            <use xlinkHref="#icon-arrow-select"/>
          </svg>
        </span>
        <ul className={`${isOpen ? `places__options--opened` : ``} places__options places__options--custom`}>
          {SortOptions.map((option) => {
            return <Option
              onSelectOption={onSelectOption}
              onSetTypeSort={onSetTypeSort}
              isSelected={currentSortType === option.sortType}
              sortType={option.sortType}
              name={option.name}
              key={`option-${option.sortType}`}
            />;
          })}
        </ul>
      </>
    );
  }
}

Sort.propTypes = {
  currentSortType: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
  labelName: PropTypes.string.isRequired,
  onOpenSorting: PropTypes.func.isRequired,
  onSelectOption: PropTypes.func.isRequired,
  onSetTypeSort: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onSetTypeSort: (value) => dispatch(ActionCreator.setSortType(value)),
});

export {Sort};
export default connect(null, mapDispatchToProps)(Sort);
