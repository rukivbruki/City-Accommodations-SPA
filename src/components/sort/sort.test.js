import React from "react";
import renderer from "react-test-renderer";
import {Sort} from "./sort";

it(`Sort component render correct`, () => {
  const sortTemplate = renderer
    .create(<Sort
      currentSortType={``}
      isOpen={true}
      labelName={``}
      onOpenSorting={jest.fn()}
      onSelectOption={jest.fn()}
      onSetTypeSort={jest.fn()}
    />)
    .toJSON();

  expect(sortTemplate).toMatchSnapshot();
});
