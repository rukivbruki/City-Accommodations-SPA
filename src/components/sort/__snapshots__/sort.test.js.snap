// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component render correct 1`] = `
Array [
  <span
    className="places__sorting-type"
    onClick={[MockFunction]}
    tabIndex="0"
  >
    
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>,
  <ul
    className="places__options--opened places__options places__options--custom"
  >
    <li
      className=" places__option"
      data-value="POPULAR"
      onClick={[Function]}
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className=" places__option"
      data-value="TO_HIGH"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className=" places__option"
      data-value="TO_LOW"
      onClick={[Function]}
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className=" places__option"
      data-value="TOP_RATED"
      onClick={[Function]}
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>,
]
`;
